src/Servers/Kestrel/Core/test/Http1/Http1OutputProducerTests.cs

```diff
@@ -54,6 +54,61 @@ public async Task WritesNoopAfterConnectionCloses()
         }
     }
 
+    [Fact]
+    public async Task FlushAsync_OnDisposedSocket_ReturnsResultWithIsCompletedTrue()
+    {
+        var pipeOptions = new PipeOptions
+        (
+            pool: _memoryPool,
+            readerScheduler: Mock.Of<PipeScheduler>(),
+            writerScheduler: PipeScheduler.Inline,
+            useSynchronizationContext: false
+        );
+
+        var socketOutput = CreateOutputProducer(pipeOptions);
+
+        await socketOutput.WriteDataAsync(new byte[] { 1, 2, 3, 4 }, default);
+        var successResult = await socketOutput.FlushAsync();
+        Assert.False(successResult.IsCompleted);
+
+        // Close
+        socketOutput.Dispose();
+
+        await socketOutput.WriteDataAsync(new byte[] { 1, 2, 3, 4 }, default);
+        var cancelResult = await socketOutput.FlushAsync();
+        Assert.True(cancelResult.IsCompleted);
+
+        socketOutput.Pipe.Writer.Complete();
+        socketOutput.Pipe.Reader.Complete();
+    }
+    [Fact]
+    public async Task FlushAsync_OnSocketWithCanceledPendingFlush_ReturnsResultWithIsCanceledTrue()
+    {
+        var pipeOptions = new PipeOptions
+        (
+            pool: _memoryPool,
+            readerScheduler: Mock.Of<PipeScheduler>(),
+            writerScheduler: PipeScheduler.Inline,
+            useSynchronizationContext: false
+        );
+
+        var socketOutput = CreateOutputProducer(pipeOptions);
+
+        await socketOutput.WriteDataAsync(new byte[] { 1, 2, 3, 4 }, default);
+        var successResult = await socketOutput.FlushAsync();
+        Assert.False(successResult.IsCanceled);
+
+        // Close
+        socketOutput.CancelPendingFlush();
+
+        await socketOutput.WriteDataAsync(new byte[] { 1, 2, 3, 4 }, default);
+        var cancelResult = await socketOutput.FlushAsync();
```

Comment:
```suggestion
        var cancelResult = await socketOutput.WriteDataToPipeAsync(new byte[] { 1, 2, 3, 4 }, default);
```

This was effectively double flushing before because it was calling the internal `WriteDataAsync` API that's called by `HttpResponse.Body` rather than `WriteDataToPipeAsync` that's called by `HttpRessponse.BodyWriter`.

I wonder if this is why you weren't seeing canceled `FlushResult`s  before this change. The second flush shouldn't be canceled as I mentioned earlier.